README.md

```diff
@@ -447,6 +448,181 @@ you can create your own Configuration(s) and compose them with Config's ++ opera
+
+For that we need to build a Remote Bit Bang enabled emulator, this could be any of the listed configurations in `rocket-chip/src/main/scala/system/Configs.scala`. Basically all we need is to extend our configuration with JtagDTMSystem that will deal with program loading and interface with the debugger. In the following example we added this extention to the DefaultConfig one.
+
+    class DefaultConfigRBB extends Config(
+    new WithJtagDTMSystem ++ new WithNBigCores(1) ++ new BaseConfig)
+
+    class QuadCoreConfigRBB extends Config(
+    new WithJtagDTMSystem ++ new WithNBigCores(4) ++ new BaseConfig)
+
+To build the configuration we use the command:
+
+    rocket-chip$ cd emulator
+    emulator$ CONFIG=DefaultConfigRBB make
+
+By default the emulator is generated under the name `emulator-freechips.rocketchip.system-DefaultConfigRBB`.
+
+### 2) Compiling the executable
+
+We suppose that `helloworld` is our program, you can use `crt.S`, `syscalls.c` and the linker script `test.ld` to construct your own program, check examples stated in [riscv-tests](https://github.com/riscv/riscv-tests).
+
+In our case we will use the following example:
+
+```
+char text[] = "Vafgehpgvba frgf jnag gb or serr!";
+
+// Don't use the stack, because sp isn't set up.
+volatile int wait = 1;
+
+int main()
+{
+    while (wait)
+        ;
+
+    // Doesn't actually go on the stack, because there are lots of GPRs.
+    int i = 0;
+    while (text[i]) {
+        char lower = text[i] | 32;
+        if (lower >= 'a' && lower <= 'm')
+            text[i] += 13;
+        else if (lower > 'm' && lower <= 'z')
+            text[i] -= 13;
+        i++;
+    }
+
+    while (!wait)
+        ;
+}
+```
+Do not forget to compile with `-g -Og` flags to provide debugging support just as explained [here](https://github.com/riscv/riscv-isa-sim#debugging-with-gdb).
+
+### 3) Launch the emulator
+
+We can then launch the emulator with
+
+    ./emulator-freechips.rocketchip.system-DefaultConfigRBB +jtag_rbb_enable=1 --rbb-port=9823 helloworld
+	This emulator compiled with JTAG Remote Bitbang client. To enable, use +jtag_rbb_enable=1.
+	Listening on port 9823
+	Attempting to accept client socket
+
+### 4) Launch OpenOCD
+
+We first need to ensure the OpenOCD has been generated. Generally it is in `$(RISCV)/bin/openocd`. We then define a configuration file that is going to define the RBB port we will use, which is in our case `9823`.
+
+    $ cat cemulator.cfg 
+    interface remote_bitbang
+    remote_bitbang_host localhost
+    remote_bitbang_port 9823
+
+    set _CHIPNAME riscv
+    jtag newtap $_CHIPNAME cpu -irlen 5
+
+    set _TARGETNAME $_CHIPNAME.cpu
+    target create $_TARGETNAME riscv -chain-position $_TARGETNAME
+
+    gdb_report_data_abort enable
+
+    init
+    halt
+
+Then we launch OpenOCD in another terminal using the command
+    
+    $ openocd -f ./cemulator.cfg
```

Comment:
please use the full path `$(RISCV)/bin/openocd`. This is a common mistake.

Reply:
You can also suggest adding the `-d` flag for debug info

Reply:
Done.